File: 2020/day-1/part2.rb
#!/usr/bin/env ruby
require 'open-uri'

numbers = open('input.txt').read.split.map(&:to_i)

numbers.each do |x|
numbers.each do |y|
numbers.each do |z|
  next if x == y || y == z || z == x
  puts x * y * z if x + y + z == 2020
end
end
end

File: 2020/day-3/part2.py
#!/usr/bin/env python3

shifts = [
  { "x": 1, "y": 1 },
  { "x": 3, "y": 1 },
  { "x": 5, "y": 1 },
  { "x": 7, "y": 1 },
  { "x": 1, "y": 2 }
]

filepath = 'input.txt'
map = open(filepath).read().split('\n')

height = len(map)
width = len(map[0])

trees_multiplication = 1

for shift in shifts:
  trees_cnt = 0
  x = 0
  y = 0

  while y < height:
    if map[y][x] == '#':
      trees_cnt += 1
    x = (x + shift["x"]) % width
    y += shift["y"]

  trees_multiplication *= trees_cnt
  print(trees_cnt)

print(trees_multiplication)